ts/private/ts_validate_options.bzl

```diff
@@ -23,6 +24,20 @@ def _validate_options_impl(ctx):
     # We make it a .d.ts file so we can plumb it to the deps of the ts_project compile.
     marker = ctx.actions.declare_file("%s.optionsvalid.d.ts" % ctx.label.name)
 
+    # Provider validation
+    if not ctx.attr.allow_js:
+        for d in ctx.attr.deps:
+            if not d[JsInfo].declarations:
+                fail("""\
+ts_project '{0}' does not contain any declarations (.d.ts or other type-check files).
```

Comment:
@alexeagle I think assuming this dep is a `ts_project` might be confusing in the cases it is not such as the CI failure I was getting (depending on a `js_library` that has no declarations).

Maybe... "ts_project {1} dependency {0} does not contain ..."?

Reply:
I'm fixing things up atm and will do it